experiments/experiment.go

```diff
@@ -438,13 +446,22 @@ func (ec *experimentContext) assessTemplates() (v1alpha1.AnalysisPhase, string)
 func (ec *experimentContext) assessAnalysisRuns() (v1alpha1.AnalysisPhase, string) {
 	worstStatus := v1alpha1.AnalysisPhaseSuccessful
 	message := ""
+
+	requiredAnalysisRunCompleted := false
 	for _, a := range ec.ex.Spec.Analyses {
 		as := experimentutil.GetAnalysisRunStatus(*ec.newStatus, a.Name)
+		if as.RequiredForCompletion && as.Phase == v1alpha1.AnalysisPhaseSuccessful {
+			requiredAnalysisRunCompleted = true
```

Comment:
I think theres a bug here. This will set `requiredAnalysisRunCompleted` to be true as soon as **_any one_** of the required runs completes, but not all. I think this should be reversed to set it to true and set it to false as soon as one required run is still not complete.

Reply:
fixed